### DISCUSSION TITLE
Content for Milestone 2 video

### DISCUSSION THREAD

**Discussion by VietAnh1010** [joulev/ezkomment#144] [General]
Start brainstorming for ideas. Some suggestions are appreciated.

I will try to make the content as concise as possible. A lengthy but boring video is a waste of time to watch &ndash; and we must avoid that.

**Comment by joulev** [comment]
The main content can be taken from the README. I think we only need to reformat and summarise it in some way to a presentation.

**Comment by joulev** [comment]
* One slide for problem statement and motivation (30s)
* 1-2 slide(s) for a brief description of the features (1-1.5m)
* 1 slide about what we have done: CRUD sites, CRUD pages, etc. (1m)
* Walkthrough of the product (1-1.5m)
* 1-2 slide about [app design and implementation](https://test.joulev.dev/orbital/ms2-readme#application-design-and-implementation) (1.5m)
* 1 slide for problems faced (30s)
* 1 slide for plans and conclusion (30s-1m)

Guess that'll be enough.

**Reply by VietAnh1010** [reply]
> One slide for problem statement and motivation (30s)

30s for this is good.

> 1-2 slide(s) for a brief description of the features (1-1.5m)

I would integrate this with the walkthrough.

> 1 slide about what we have done: CRUD sites, CRUD pages, etc. (1m)
> Walkthrough of the product (1-1.5m)

Watching the MS2 videos of the previous years, it seems like for MS2 the teams mostly concentrated on displaying the features that have been implemented (These videos are not very good, too slow, and do not seem very interesting at all).

We will do a walkthrough with the current features, from when the user first created their account to when they delete their account. During which, the user explores the features of the app. This should take about 2m - 2.5m.

Automating this walkthrough is preferable.

> 1-2 slides about app design and implementation (1.5m)
> 1 slide for plans and conclusion (30s-1m)

Agree with these two. But I prefer to move the planned features from `a brief description of the features` to this section, so the watcher can clearly understand what is the implemented features and what is the planned features to look out for.

For problems faced, I think leaving it in the README is better.

Total time should be around 5m-5m30s. And we can increase the duration if the current plan turns out to be too restricted on time.

**Reply by joulev** [reply]
> I would integrate this with the walkthrough.

As in, all planned features. We haven't implemented them, so we can't have a walkthrough for them.

> But I prefer to move the planned features from a brief description of the features to this section, so the watcher can clearly understand what is the implemented features and what is the planned features to look out for.

From the README you can see the features description and the application design have few things in common. I think it's a good idea to separate them.

> For problems faced, I think leaving it in the README is better.

Actually it looks like that part is *required* in the video...? Anyway it's short, we can even do it in 15s, so shouldn't be a big deal.

**Reply by VietAnh1010** [reply]
> Actually it looks like that part is required in the video...?

Not sure, but I did not see any team address that in their videos. One example: https://www.youtube.com/watch?v=WsgsoqqXkw8.

**Reply by joulev** [reply]
Okay approved, we will remove that section. Oh-my-goodness 13 minutes (looking back to us though, our README is 24 pages)

**Comment by joulev** [comment]
Any updates on the video? Also please [update your project log](https://github.com/joulev/ezkomment/edit/main/pages/orbital/log-vietanh.json).

**Reply by VietAnh1010** [reply]
Sorry, today I have to go out. Will complete the video content and the project log now.

**Reply by VietAnh1010** [reply]
> Problem statement and motivation (30s)

I think we reuse the slides from the MS1 video here.

> A brief description of the features (30s)

And mainly focus on two aspects:
- Ease of use (if the user just wants to build a simple website and add a comment section).
- Maximum customizability for power users.

I think we can also reuse the slides from MS1 video here?

> Walkthrough of the product

- First, the user creates an account (we must clearly show the intention that we do not support the traditional password login) (15s).
- The landing page is initially empty. We will guide them on how to create a site and emphasize the purpose of the site. (20s-30s).
- The user can also configure their site (change name, upload photo, etc.). (20s)
- After uploading the site icon, the user will create a page. (15s)
- The user now goes into the page. It can be embedded into another website.
- If possible, we can write some simple HTML and embed the page to demonstrate. (15s) Then add some comment, and they should appear on the page. (10s-15s)
- The user can toggle the auto approve button. (10s) (I am not sure whether we should include this into the video yet as we haven't completed the auto-approve feature).
- If the user doesn't need the page/site/account anymore, they can: (20-30s)
  - Delete the page, with all of their comments. OR
  - Delete the site, with all the pages and comments inside it. OR
  - Delete the entire account.

The walkthrough is completed here.

> App design and implementation

- The reasons for choosing Next.js Typescript. (30-45s).
- The reasons for choosing Firebase over SQL database (Firebase is relatively simple but reliable, and offers multiple products at the same time: Authentication, Firestore, Storage). (30s)
- The reasons for choosing Tailwind CSS (no one can stand the CSS mess). (15s-20s)
- Then demonstrate the reasons why we choose to deploy on Vercel (superb for front-end, nearly no latency for backend with their serverless functions). (20s-30s)
- F*ck OOP, no SOLID or singleton pattern because our code is just functions and data. And Typescript module is good enough to separate the code (Not sure whether we should include this).

> Plans and conclusion
- Everything we planned to complete for MS3. (30s)

**Comment by joulev** [comment]
<img width="482" alt="image" src="https://user-images.githubusercontent.com/44609036/175908969-eda1f4db-7265-4243-a696-6fbf10eecafa.png">

Ah shoot, I think I overdid it.

**Reply by joulev** [reply]
https://www.figma.com/proto/zGVcNdL5BYWkxUfJ1XJVy8/ezkomment?page-id=0%3A1&node-id=323%3A49&viewport=140%2C251%2C0.04&scaling=min-zoom&starting-point-node-id=323%3A49

This is the slides, you can give it a try.

**Reply by joulev** [reply]
Never mind I uploaded a 10-min version, we don’t need to care about this anymore.

**Reply by VietAnh1010** [reply]
Oh my god, 22 minutes 🤣

**Reply by joulev** [reply]
> Oh my god, 22 minutes 🤣

22 pages of README, accompanied by 22 minutes of a boring presentation.

Perfectly balanced.

**Reply by VietAnh1010** [reply]
The video is still 11 minutes, much longer than I expected.

And it would be better if both member present the video. You can just memtion me, I will send you my voice recording.

**Reply by joulev** [reply]
It's pretty hard to do so though, especially as we have to manually edit the video (and burn the laptop of either member to ashes in the process).

Other teams can do so because they make the recording together. We are literally 300km away from each other right now, that's simply impossible. (Or possible by Zoom but that will only make the already-shitty audio even shittier.)

Don't worry about it. Most of the time I spent is actually me playing around with Figma. The recording didn't take a lot of time.
